RecoTracker/MkFitCore/src/MkBuilder.cc

```diff
@@ -1115,6 +1118,55 @@ namespace mkfit {
         mkfndr->copyOutParErr(eoccs.refCandidates_nc(), end - itrack, false);
       }
 
+      const int theEndOverlapper = seed_cand_overlap_idx.size();
+
+      for (int itrack = 0; itrack < theEndOverlapper; itrack += NN) {
+        const int end = std::min(itrack + NN, theEndOverlapper);
+
+        mkfndr->inputTracksAndHits(eoccs.refCandidates(), layer_of_hits, seed_cand_overlap_idx, itrack, end, true);
+
+        mkfndr->updateWithLoadedHit(end - itrack, fnd_foos);
+
+        mkfndr->copyOutParErr(eoccs.refCandidates_nc(), end - itrack, false);
+
+        mkfndr->inputOverlapHits(layer_of_hits, seed_cand_overlap_idx, itrack, end);
+
+        // XXXX Could also be calcChi2AndUpdate(), then copy-out would have to be done
+        // below, choosing appropriate slot (with or without the overlap hit).
+        // Probably in a dedicated MkFinder copyOutXyzz function.
+        mkfndr->chi2OfLoadedHit(end - itrack, fnd_foos);
+
+        for (int ii = itrack; ii < end; ++ii) {
+          const int fi = ii - itrack;
+          TrackCand &tc = eoccs[seed_cand_overlap_idx[ii].seed_idx][seed_cand_overlap_idx[ii].cand_idx];
+
+          // XXXX For now we DO NOT use chi2 as this was how things were done before the post-update
+          // chi2 check. To use it we should retune scoring function (might be even simpler).
+          if (mkfndr->m_FailFlag[fi] == 0 && mkfndr->m_Chi2[fi] >= 0.0f && mkfndr->m_Chi2[fi] <= 60.0f) {
```

Comment:
I'd like to propose to make this configurable. We have `chi2CutOverlap` parameter in iter config already, which is currently unused; and I guess it can be reused here.
the logic could probably be 
```suggestion
          if (m_iteration_params->chi2CutOverlap < 0 || (mkfndr->m_FailFlag[fi] == 0 && mkfndr->m_Chi2[fi] >= 0.0f && mkfndr->m_Chi2[fi] <= m_iteration_params->chi2CutOverlap)) {
```

Reply:
the default could be set to `-1` in all iterations jsons for now

Reply:
We also have pT cut for overlap ... it used to be applied so only tacks with pT > 1 GeV would pick up overlaps.

Default in jsons now seem to be 3.5 for chi2 and 0 for pT ... neither of them being used now.

Do you want me to just commit the above suggestion for now then?

Reply:
uhm, `pTCutOverlap` is used in RecoTracker/MkFitCore/src/CandCloner.cc; it does nothing because the value is zero. I can see it could become useful/reconsidered to change from 0 and pt being relatively well defined, I think it's fine where it is.